src/hw3/hw3_3.py

```diff
@@ -0,0 +1,44 @@
+# https://www.python.org/dev/peps/pep-0570/#logical-ordering
+# Positional-only parameters also have the (minor) benefit of enforcing some logical order when
+# calling interfaces that make use of them. For example, the range function takes all its
+# parameters positionally and disallows forms like:
+
+# range(stop=5, start=0, step=2)
+# range(stop=5, step=2, start=0)
+# range(step=2, start=0, stop=5)
+# range(step=2, stop=5, start=0)
+
+# at the price of disallowing the use of keyword arguments for the (unique) intended order:
+
+# range(start=0, stop=5, step=2)
+"""
+Write a function that accept any sequence (list, string, tuple) of unique values and then
+it behaves as range function:
+
+
+import string
+
+
+assert = custom_range(string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(string.ascii_lowercase, 'g', 'p') == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+
+"""
+import inspect
+
+
+def custom_range(value, start=inspect._empty, stop=inspect._empty, step=inspect._empty):
```

Comment:
You are bypassing the test again, but not solving the problem. The point of this exercise is to create a function, that doesn't take keyword arguments - only positional ones.
You can do it by creating variable length `*args`. And inside your function you can check - if number of args is 1 - then start should be 0, stop ..., step ... 
And similarly for 2, 3, or 4 arguments

Reply:
thx